Content.Client/DragDrop/DragDropSystem.cs

```diff
@@ -18,479 +16,525 @@
 using Robust.Shared.Configuration;
 using Robust.Shared.Input;
 using Robust.Shared.Input.Binding;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
 using Robust.Shared.Prototypes;
 using Robust.Shared.Utility;
 using DrawDepth = Content.Shared.DrawDepth.DrawDepth;
 
-namespace Content.Client.DragDrop
+namespace Content.Client.DragDrop;
+
+/// <summary>
+/// Handles clientside drag and drop logic
+/// </summary>
+[UsedImplicitly]
+public sealed class DragDropSystem : SharedDragDropSystem
 {
+    [Dependency] private readonly IStateManager _stateManager = default!;
+    [Dependency] private readonly IInputManager _inputManager = default!;
+    [Dependency] private readonly IEyeManager _eyeManager = default!;
```

Comment:
death to IEyeManager, long live getting the control under the current mouse position, checking if its a viewport, then using that viewports screen to map coordinates.

Actually now that I think about it that should probably be some helper method provided by IEyeManager or IInputManager, but that requires an engine PR, so maybe leave it as is for now.